contracts/instance/ProductStore.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import {AccessManagedUpgradeable} from "@openzeppelin/contracts-upgradeable/access/manager/AccessManagedUpgradeable.sol";
+
+import {IInstance} from "./IInstance.sol";
+import {IPolicy} from "./module/IPolicy.sol";
+
+import {Key32} from "../type/Key32.sol";
+import {NftId} from "../type/NftId.sol";
+import {ClaimId} from "../type/ClaimId.sol";
+import {ObjectType} from "../type/ObjectType.sol";
+import {StateId} from "../type/StateId.sol";
+import {PayoutId} from "../type/PayoutId.sol";
+import {BalanceStore} from "./base/BalanceStore.sol";
+import {BaseStore} from "./BaseStore.sol";
+import {Blocknumber} from "../type/Blocknumber.sol";
+import {KeyId} from "../type/Key32.sol";
+import {ObjectCounter} from "./base/ObjectCounter.sol";
+import {ObjectLifecycle} from "./base/ObjectLifecycle.sol";
+import {ObjectType, POLICY} from "../type/ObjectType.sol";
+
+
+contract ProductStore is
+    AccessManagedUpgradeable, 
+    BalanceStore,
+    BaseStore,
+    ObjectCounter,
+    ObjectLifecycle
+{
+    event LogProductStoreInfoCreated(ObjectType objectType, KeyId keyId, StateId state, address createdBy, address txOrigin);
```

Comment:
use object specific logs:
- LogProductStoreInfoCreated -> LogProductStorePolicyInfoCreated
- objectType can be removed
- instead of KeyId keyId, use NftId policyNftId